### PR TITLE
tests: test fleetd with extra configuration options

### PR DESCRIPTION
**Pull request by kayrus**
covers #1418

/cc @tixxdz

### CODE REVIEW

**Comment by tixxdz** [comment]
cc @hectorj2f
 
@kayrus thank you! so we wait for replies on this https://github.com/coreos/fleet/pull/1418#issuecomment-211378197  we see, we test it, and if it's hard to achieve note 1) of that comment, we merge this one and we follow up with PR #1418 

Thanks!

File: functional/test [hunk 1]
```diff
@@ -24,28 +49,19 @@ fi
...
-echo "# TOTAL: $total"
-echo "# PASS: $pass"
-echo "# FAIL: $fail"
-echo ""
-echo "See ${CDIR}/functional-tests.log for the detailed output."
-exit $TESTS_RETURN_CODE
+TESTS_RETURN_CODE_1=${PIPESTATUS[0]}
+
+# Run tests with cache enable on fleetd side
+export TEST_ENV="enable_unitstate_cache=true"
```
Comment (tixxdz) [inline functional/test]: Could you rename from TEST_ENV to FLEETD_TEST_ENV ?

**Comment by tixxdz** [comment]
@kayrus thanks for the PR!

Currently the output is:
`-- PASS: TestKnownHostsVerification (enable_unitstate_cache=true) (6.02s)`

Would it be better to switch to:
`-- PASS: TestKnownHostsVerification (fleet.conf=[enable_unitstate_cache=true]) (6.02s)`
?

lgtm overall! tests pass, so waiting for @hectorj2f before proceeding.

**Comment by dongsupark** [comment]
BTW this PR needs to be rebased too.

**Comment by dongsupark** [comment]
I rebased this PR, as @kayrus is on vacation. Also I renamed a variable from TEST_ENV to FLEET_TEST_ENV.

**Comment by dongsupark** [comment]
Oops, I had missed a change in ``functional/util/config.go``. So I updated it by renaming ``TEST_ENV`` to ``FLEETD_TEST_ENV``.

**Comment by dongsupark** [comment]
Fixed more bugs.
* Fixed a wrong pattern : from ``PASS|SKIP|SKIP`` to ``PASS|SKIP|FAIL``.
* Added a prefix ``fleet.conf=`` to output strings, like @tixxdz suggested. 
* Disabled the second round of test, as unit state cache is going to be obsolete.

**Comment by dongsupark** [comment]
If no one objects, I'll merge it tomorrow.

**Pull request merged**